enhancements/aws-permissions-check-bypass.md

```diff
@@ -0,0 +1,269 @@
+to satisfy a CredentialsRequest.
+
+## Design Details
+
+### Test Plan
+
+Ideally, e2e coverage of installing OpenShift in an AWS account with SCP
+permissions defined in a way that would otherwise fail without bypassing
+permissions checking. Acceptably, simply running the installation with the
+environment variable defined to force bypassing permissions checking.
+
+We should also consider the case where OpenShift was installed in an AWS account
+without SCPs defined, but the account is then migrated to an environment where
+SCPs are now defined post-installation. CCO should be able to recover by an
+admin defining the ConfigMap to force CCO into `mint` or `passthrough` mode as
+appropriate. CCO will eventually settle into a functioning state (assuming the
+credentials have sufficient permissions).
+
+### Graduation Criteria
+
+**Note:** *Section not required until targeted at a release.*
+
+Define graduation milestones.
+
+These may be defined in terms of API maturity, or as something else. Initial proposal
+should keep this high-level with a focus on what signals will be looked at to
+determine graduation.
+
+Consider the following in developing the graduation criteria for this
+enhancement:
+- Maturity levels - `Dev Preview`, `Tech Preview`, `GA`
+- Deprecation
+
+Clearly define what graduation means.
+
+#### Examples
+
+These are generalized examples to consider, in addition to the aforementioned
+[maturity levels][maturity-levels].
+
+##### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- End user documentation, relative API stability
+- Sufficient test coverage
+- Gather feedback from users rather than just developers
+
+##### Tech Preview -> GA 
+
+- More testing (upgrade, downgrade, scale)
+- Sufficient time for feedback
+- Available by default
+
+**For non-optional features moving to GA, the graduation criteria must include
+end to end tests.**
+
+##### Removing a deprecated feature
+
+- Announce deprecation and support policy of the existing feature
+- Deprecate the feature
+
+### Upgrade / Downgrade Strategy
+
+N/A. A running cluster whose AWS account is migrated to an organization that
+does define SCPs will start to error.
+
+During cluster runtime, enabling/disabling the bypassing of permissions
+simulations can be controlled through the contents of the ConfigMap.
+
+### Version Skew Strategy
+
+The installer generates the configmap for CCO, and it will have to stay in sync
+with the format for delivering this `mint` vs `passthrough` information if/when
+CCO moves away from using a ConfigMap.
+
+## Implementation History
+
+## Drawbacks
+
+Moving away from pre-flight permissions checks pushes out the time for when
+someone attempting to install OpenShift will get feedback on success/failure.
+The pre-flight checks have not exposed OpenShift to needing to bubble up
+appropriate information when certain types of cloud API errors are encountered.
```

Comment:
@jeremyeder what will SD do here? Would you always be forcing a mode with Hive? Running some other pre-flight routine before invoking Hive?